scripts/pancancer.py

```diff
@@ -0,0 +1,341 @@
+                if 'http_workflow' in workflow_details:
+                    paramsData['http_workflows'] = workflow_details['http_workflow']
+                if 's3_workflows' in workflow_details:
+                    paramsData['s3_workflows'] = workflow_details['s3_workflows']
+                if 'containers' in workflow_details:
+                    paramsData['containers'] = workflow_details['containers']
+                if 's3_containers' in workflow_details:
+                    paramsData['s3_containers'] = workflow_details['s3_containers']
+                if 'http_containers' in workflow_details:
+                    paramsData['http_containers'] = workflow_details['http_containers']
+
+                paramsData['lvm_device_whitelist'] = workflow_details['lvm_devices']
+                paramsData['workflow_name'] = workflow_name
+
+            # Now write the params.json file.
+            with open('/home/ubuntu/params.json','w+') as params_file:
+                params_file.write(str(json.dumps(paramsData,sort_keys=True, indent=4) ))
+
+            # Update the youxia config file with the correct AMI and instance-type
+            config = configparser.ConfigParser()
+            config.read('/home/ubuntu/.youxia/config')
+            config['deployer']['instance_type']=workflow_details['instance-type']
+            config['deployer']['ami_image']=workflow_details['ami_id']
+            with open('/home/ubuntu/.youxia/config','r+') as youxia_configfile:
+                config.write(youxia_configfile,space_around_delimiters=True)
+
+            subprocess.check_call(generator_cmd.split(' '))
+
+###
+
+class Reports(Command):
+    "This will generate reports on the command line."
+    log = logging.getLogger(__name__)
+    def get_parser(self,prog_name):
+        parser = super(Reports,self).get_parser(prog_name)
+        subparser = parser.add_subparsers(title='subcommands',help='reporting subcommands: gather, info, jobs, provisioned, status, help',dest='report_subcmd')
+        subparser.required=True
+        subparser.add_parser('gather',help='gathers the last message sent by each worker and displays the last line of it')
+        subparser.add_parser('info',help='retrieves detailed information on provisioned instances')
+        subparser.add_parser('jobs',help='retrieves detailed information on jobs')
+        subparser.add_parser('provisioned',help='retrieves detailed information on provisioned instances')
+        subparser.add_parser('status',help='retrieves configuration and version information on arch3')
+        subparser.add_parser('youxia',help='ask youxia for all information on instances known to the cloud APIs that are configured')
+        subparser.add_parser('help',help='Prints a help message for the reporting system.')
+        return parser
+
+    def take_action(self, parsed_args):
+        subcmd = vars(parsed_args)['report_subcmd']
+        cmd_str=''
+        if subcmd!='help':
```

Comment:
Why not just an else?  Why elif?  You can miss a pass through here.
You can also use 'is not' instead of != to improve readability.